.github/workflows/php.yml

```diff
@@ -0,0 +1,56 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    # ubuntu contains php by default (2022)
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clone matomo-org/matomo
+      uses: actions/checkout@v2
+      with:
+        repository: matomo-org/matomo
+        submodules: true
+        #ref: 4.7.x
+        
+
+    - name: Validate composer.json and composer.lock
+      run: composer validate --strict
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
+      run: composer install --prefer-dist --no-progress
+      
+      
+      
+    - name: Clone plugin repo into plugins subdirectory of the matomo installation
+      uses: actions/checkout@v2
+      with:
+        repository: wmde/CampaignVisitorsByTime
+        path: plugins/
+        #ref: TODO
+      
+    - name: Run unit tests
+      run: ./console tests:run plugins/CampaignVistorsByTime/tests/Unit/DataProcessorTest.php
+      
+    - name: Run phpcs
+      run: phpcs
```

Comment:
This command might need a `vendor/bin/` prefix. It will then run phpcs for the whole matomo project, with the [Matomo coding style](https://github.com/matomo-org/matomo/blob/4.x-dev/phpcs.xml) probably throwing a lot of errors for our plugin because we use [some old version of the FundraisingTech Coding style](https://github.com/wmde/CampaignVisitorsByTime/blob/master/phpcs.xml), with a dependency (slevomat/coding-standard) that's not in the Matomo dependencies.